app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -119,7 +121,11 @@ def index
     end
 
     def index_relation
-      query = Repository.readable
+      if ::Foreman::Cast.to_bool params[:archived]
+        query = params[:docker_tag_id].nil? ? Repository.readable : DockerMetaTag.find(params[:docker_tag_id]).repositories.readable
+      else
+        query = params[:docker_tag_id].nil? ? Repository.non_archived.readable : DockerMetaTag.find(params[:docker_tag_id]).repositories.non_archived.readable
```

Comment:
I realized I need to re-work the default behavior here since it's different from normal.